modules/c++/math/include/math/Sign.h

```diff
@@ -0,0 +1,49 @@
+/* =========================================================================
+ * This file is part of math-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2016, MDA Information Systems LLC
+ *
+ * math-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef __MATH_SIGN_H__
+#define __MATH_SIGN_H__
+
+namespace math
+{
+/*!
+ * Find sign of input, expressed as -1, 0, or 1
+ * \param val A signed number
+ * \return 1 if val is positive, -1 if negative, 0 otherwise
+ */
+template <typename T>
+int sign(const T val)
```

Comment:
Removed.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Thursday, August 25, 2016 9:53 PM
To: mdaus/coda-oss <coda-oss@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [mdaus/coda-oss] Add sign function (#159)


In modules/c++/math/include/math/Sign.h<https://github.com/mdaus/coda-oss/pull/159#discussion_r76354121>:

> + * see <http://www.gnu.org/licenses/>.

> + *

> + */

> +

> +#ifndef __MATH_SIGN_H__

> +#define __MATH_SIGN_H__

> +

> +namespace math

> +{

> +/*!

> + * Find sign of input, expressed as -1, 0, or 1

> + * \param val A signed number

> + * \return 1 if val is positive, -1 if negative, 0 otherwise

> + */

> +template <typename T>

> +int sign(const T val)

Passing in by value so no need for const

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/mdaus/coda-oss/pull/159/files/f261a6fbd6b7ba4ba4ef1a39878a696a973490ed#r76354121>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYCxeM76QRUFHyv5iO2uXWBN4t8kVks5qjkcPgaJpZM4Jr7QN>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.